Source/WebCore/page/Page.h

```diff
@@ -555,7 +555,11 @@ class Page : public RefCountedAndCanMakeWeakPtr<Page>, public Supplementable<Pag
     void setTabKeyCyclesThroughElements(bool b) { m_tabKeyCyclesThroughElements = b; }
     bool tabKeyCyclesThroughElements() const { return m_tabKeyCyclesThroughElements; }
 
-    WEBCORE_EXPORT std::optional<FrameIdentifier> findString(const String&, FindOptions, DidWrap* = nullptr);
+    struct FindStringData {
+        std::optional<FrameIdentifier> frameIdentifer;
+        std::optional<SimpleRange> range;
+    };
+    WEBCORE_EXPORT FindStringData findString(const String&, FindOptions, DidWrap* = nullptr);
```

Comment:
It seems like we could make `findString` return an `optional<FindStringData>`, and make the members of `FindStringData` non-optional?

Reply:
Is that possible?

Reply:
Probably not, but if we leave it in this form, we can remove the out-parameter as pointed out by Richard below, which I think is also valuable.

Reply:
(I agree more with Aditya than my own suggestion)